### ISSUE TITLE
The Thing with "special" Money

### ISSUE THREAD

**Issue by Avankziar** [Hempfest/Hemponomics#5]
I think I can say that we all know it. That next to the "normal" money (Dollar, Euro, ect.) something like a special currency exists.
Be it tokens, coins or whatever.

Now when I talk about a currency system and have Vault in mind, my question would be why isn't there the same for "special" currencies? Aka a second interface which is similar to the normal currencies but is a different class/interface.

There you could then also make methods like Withdraw and Deposit, Multiple special currencies support which can be distinguished by a key (String or NamespacedKey or whatever).

For example:
EconomyAction createAccount(AccountType type, OfflinePlayer player, String accountId, String world, double amount);
//It's already in there

SpecialEconomyAction createAccount(SpecialCurrencyType type, AccountType type, OfflinePlayer player, String accountId, String world, double amount);
//And that would be, for example, the one for the special currencies. Specified this is a Example ;)

If bank accounts are suitable for something like this, I would rather say no. But that would be in the eye of the beholder.

It should be emphasized again that multiple currencies should be supported here directly. I mean simultaneously, so that tokens or coins could be used at the same time during the game. 
Most of the time, you get something different for tokens than for coins, etc. 
In other words, tokens and coins are both special currencies, but coins is more special, if you understand what I mean^^.
